File: crates/enforcer-memory/tests/fixtures/memory/lang_java/Widget.java
package com.example.widget;

import java.util.List;
import java.util.ArrayList;

public interface Drawable {
    String draw();
}

public abstract class Shape implements Drawable {
    public abstract String draw();
}

public class Widget extends Shape {
    public static final int MAX_WIDGETS = 10;

    private String name;

    public Widget(String name) {
        this.name = name;
    }

    @Override
    public String draw() {
        return String.format("widget:%s", this.name);
    }
}

File: crates/enforcer-memory/tests/fixtures/memory/lang_go/widget.go
package widget

import (
	"fmt"
	"net/http"
)

// Base is embedded into Widget below -- best-effort INHERITS source.
type Base struct {
	ID int
}

type Widget struct {
	Base
	Name string
}

type Drawable interface {
	Draw() string
}

const MaxWidgets = 10

var registry = map[string]int{}

func NewWidget(name string) *Widget {
	return &Widget{Name: name}
}

func (w *Widget) Draw() string {
	return fmt.Sprintf("widget:%s", w.Name)
}

func RegisterRoutes(mux *http.ServeMux) {
	mux.HandleFunc("/widgets", ListWidgets)
}

func ListWidgets(w http.ResponseWriter, r *http.Request) {
	fmt.Println("listing")
}
